### PR TITLE
LV2: cleanup syscall table

### PR DESCRIPTION
**Pull request by elad335**
Remove "syscall names" table.

### CODE REVIEW

**Comment by Nekotekina** [comment]
Ok but how to verify names haven't changed

File: rpcs3/Emu/Cell/lv2/lv2.cpp [hunk 1]
```diff
@@ -81,652 +81,657 @@ static bool null_func(ppu_thread& ppu)
...
 {
 	null_func,
-	BIND_FUNC(sys_process_getpid),                          //1   (0x001)
-	BIND_FUNC(sys_process_wait_for_child),                  //2   (0x002)  ROOT
-	null_func,//BIND_FUNC(sys_process_exit),                //3   (0x003)
-	BIND_FUNC(sys_process_get_status),                      //4   (0x004)  DBG
-	BIND_FUNC(sys_process_detach_child),                    //5   (0x005)  DBG
+	BIND_SYSC(sys_process_getpid),                          //1   (0x001)
+	BIND_SYSC(sys_process_wait_for_child),                  //2   (0x002)  ROOT
+	NULL_FUNC(sys_process_exit),                //3   (0x003)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/lv2.cpp:110]: Well, made a check myself... syscall 3 should be null_func.

File: rpcs3/Emu/Cell/lv2/lv2.cpp [hunk 2]
```diff
@@ -81,652 +81,657 @@ static bool null_func(ppu_thread& ppu)
...
+	NULL_FUNC(sys_bt_register_controller),                  //590 (0x24E)  ROOT
+	NULL_FUNC(sys_bt_clear_registered_contoller),           //591 (0x24F)
+	NULL_FUNC(sys_bt_connect_accept_controller),            //592 (0x250)
+	NULL_FUNC(sys_bt_get_local_bdaddress),                  //593 (0x251)  ROOT
+	NULL_FUNC(sys_bt_hid_get_data),                         //594 (0x252)
+	NULL_FUNC(sys_bt_hid_set_report),                       //595 (0x253)
+	NULL_FUNC(sys_bt_sched_log),                            //596 (0x254)
+	NULL_FUNC(sys_bt_cancel_connect_accept_controller),     //597 (0x255)
+	NULL_FUNC(sys_bluetooth_...),                           //598 (0x256)  ROOT
+	NULL_FUNC(sys_bluetooth_...),                           //599 (0x257)  ROOT
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/lv2.cpp:609]: 598 & 599 should be null_func

File: rpcs3/Emu/Cell/lv2/lv2.cpp [hunk 3]
```diff
@@ -81,652 +81,657 @@ static bool null_func(ppu_thread& ppu)
...
+
+	BIND_SYSC(sys_rsxaudio_initialize),                     //650 (0x28A)
+	BIND_SYSC(sys_rsxaudio_finalize),                       //651 (0x28B)
+	BIND_SYSC(sys_rsxaudio_import_shared_memory),           //652 (0x28C)
+	BIND_SYSC(sys_rsxaudio_unimport_shared_memory),         //653 (0x28D)
+	null_func,//BIND_SYSC(sys_rsxaudio_create_connection),              //654 (0x28E)
+	null_func,//BIND_SYSC(sys_rsxaudio_close_connection),               //655 (0x28F)
+	null_func,//BIND_SYSC(sys_rsxaudio_prepare_process),                //656 (0x290)
+	null_func,//BIND_SYSC(sys_rsxaudio_start_process),                  //657 (0x291)
+	null_func,//BIND_SYSC(sys_rsxaudio_stop_process),                   //658 (0x292)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/lv2.cpp:669]: From 654 to 658 are missing, should be NULL_FUNC

File: rpcs3/Emu/Cell/lv2/lv2.cpp [hunk 4]
```diff
@@ -737,193 +742,196 @@ const std::array<ppu_function_t, 1024> s_ppu_syscall_table
...
-	null_func,//BIND_FUNC(sys_ss_~utoken_if)                //877 (0x36D)  ROOT
-	null_func,//BIND_FUNC(sys_ss_ad_sign)                   //878 (0x36E)
-	null_func,//BIND_FUNC(sys_ss_media_id)                  //879 (0x36F)
-	null_func,//BIND_FUNC(sys_deci3_open)                   //880 (0x370)
-	null_func,//BIND_FUNC(sys_deci3_create_event_path)      //881 (0x371)
-	null_func,//BIND_FUNC(sys_deci3_close)                  //882 (0x372)
-	null_func,//BIND_FUNC(sys_deci3_send)                   //883 (0x373)
-	null_func,//BIND_FUNC(sys_deci3_receive)                //884 (0x374)
-	null_func,//BIND_FUNC(sys_deci3_open2)                  //885 (0x375)
+	NULL_FUNC(syscall_sys_ss_get_cache_of_analog_sunset_flag), //860 (0x35C)  AUTHID
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/lv2.cpp:797]: Error in name, wrong syscall_ prefix.

File: rpcs3/Emu/Cell/lv2/lv2.cpp [hunk 5]
```diff
@@ -737,193 +742,196 @@ const std::array<ppu_function_t, 1024> s_ppu_syscall_table
...
+	BIND_SYSC(sys_ss_individual_info_manager),              //868  ROOT / DBG  AUTHID
+	NULL_FUNC(sys_ss_factory_data_manager),                 //869  ROOT
+	BIND_SYSC(sys_ss_get_console_id),                       //870 (0x366)
+	BIND_SYSC(sys_ss_access_control_engine),                //871 (0x367)  DBG
+	BIND_SYSC(sys_ss_get_open_psid),                        //872 (0x368)
+	NULL_FUNC(sys_ss_get_cache_of_product_mode),            //873 (0x369)
+	NULL_FUNC(sys_ss_get_cache_of_flash_ext_flag),          //874 (0x36A)
+	NULL_FUNC(sys_ss_get_boot_device),                      //875 (0x36B)
+	NULL_FUNC(sys_ss_disc_access_control),                  //876 (0x36C)
+	NULL_FUNC(sys_ss_~utoken_if),                           //877 (0x36D)  ROOT
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/lv2.cpp:814]: 877 should be null_func

**Review by Nekotekina — CHANGES_REQUESTED** [review]
Found some errors.

**Comment by elad335** [comment]
Fixed review comments.

**Comment by Nekotekina** [comment]
Okay, now some syscalls are marked as NULL_FUNC but functions exist
119
386
388/392 (weird)
837
841
842
862
863
867
873
874
875

**Comment by Nekotekina** [comment]
Also 659 should be null_func

**Comment by elad335** [comment]
Fixed.

**Comment by Nekotekina** [comment]
Either `constexpr auto sys_sm_ring_buzzer2 = sys_sm_ring_buzzer;` or just keep two references to the same function, idk.
sys_sm_ring_buzzer2 didn't exist in original table.

**Comment by elad335** [comment]
It doesnt affect LLVM as the name exists either way, I'll go with sys_sm_ring_buzzer2 then.

**Comment by Nekotekina** [comment]
Oh, it seems it was like this in PPUFunction table. If these are truly identical syscall, it may be fine to have same name.

**Comment by elad335** [comment]
Windows build rebuild. (failed with unrelated error)

**Review by Nekotekina — APPROVED** [review]

**Pull request merged**